### DISCUSSION TITLE
Q: Do all components need to be added to the app level before being imported?

### DISCUSSION THREAD

**Discussion by matthew-dean** [primefaces/community#1152] [PrimeVue] [answered]
In the documentation it says this:

> Usage
> Each component can be imported and registered individually so that you only bundle what you use. Import path is available in the documentation of the corresponding component.

It's not at all clear from the documentation if the following is required:

```ts
import Button from "primevue/button"

const app = createApp(App);
app.component('Button', Button);
```

OR, alternatively, if you can just do this in your components:
```vue
<script setup lang="ts">
import Button from 'primevue/button'
</script>
<template>
  <Button />
</template>
```

I think most people prefer the latter (as the latter is better for code-splitting by route), but the documentation isn't explicit on if there is something fundamentally different between importing at the component level or the app level. Is that also fine?

**Comment by tugcekucukoglu** [comment, marked as answer]
If you prefer using the second one, there is no need to import components globally. But we can suggest the first one, for example, if you want to use a component in more than one place. Again, it is not mandatory and the second implementation can be used.
